pennylane/_device.py

```diff
@@ -530,7 +548,7 @@ def batch_execute(self, circuits):
 
         return results
 
-    def execute_and_gradients(self, circuits, method="jacobian", **kwargs):
+    def execute_and_gradients(self, circuits, method="adjoint_jacobian", **kwargs):
```

Comment:
I'd be rather cautious of this change.  What is this change needed for?

Reply:
The old default value caused several failures in the `test_jacobian_products` tests

```bash
FAILED tests/interfaces/test_jacobian_products.py::TestJacobianProductResults::test_execute_jvp_basic[None-jpc4] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestJacobianProductResults::test_vjp_basic[None-jpc4] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestJacobianProductResults::test_jacobian_basic[None-jpc4] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestJacobianProductResults::test_execute_jacobian_basic[None-jpc4] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestJacobianProductResults::test_batch_execute_jvp[None-jpc4] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestJacobianProductResults::test_batch_vjp[None-jpc4] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestJacobianProductResults::test_batch_jacobian[None-jpc4] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestJacobianProductResults::test_batch_execute_jacobian[None-jpc4] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestCachingDeviceDerivatives::test_execution_caching[jpc1] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestCachingDeviceDerivatives::test_cached_on_execute_and_compute_jvps[jpc1] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestCachingDeviceDerivatives::test_cached_on_execute_and_compute_jacobian[jpc1] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
FAILED tests/interfaces/test_jacobian_products.py::TestCachingDeviceDerivatives::test_cached_on_vjps[jpc1] - AttributeError: 'DefaultQubitLegacy' object has no attribute 'jacobian'
```